app/controllers/crate/versions.js

```diff
@@ -1,21 +1,88 @@
 import Controller from '@ember/controller';
+import { inject as service } from '@ember/service';
 import { tracked } from '@glimmer/tracking';
 
+import { dropTask } from 'ember-concurrency';
+import { alias } from 'macro-decorators';
+
+function defaultVersionsContext() {
+  return { data: [], next_page: undefined };
+}
+
 export default class SearchController extends Controller {
-  queryParams = ['sort'];
+  @service store;
 
+  queryParams = ['per_page', 'sort'];
   @tracked sort;
+  @tracked per_page = 10;
```

Comment:
I think we should increase this a little bit. I don't think we need to display 1000 versions, but 100 should probably be fine? That way we don't show the pagination for the majority of crates.

Reply:
fair enough!